run.py

```
@@ -277,36 +280,66 @@ def open_browser(ip, port, use_https):
-                            proc.wait(timeout=3)
+                            
+                            # Filter: Don't kill our own runner process
+                            if proc.info['pid'] == os.getpid():
+                                continue
+                                
+                            print(f"[CLEAN] Killing stale background server process {proc.info['pid']} ({proc.info['name']}) on port {conn.laddr.port}")
+                            proc.kill() # Force kill immediately to release socket hook
+                            proc.wait(timeout=2)
             except (psutil.NoSuchProcess, psutil.AccessDenied, psutil.ZombieProcess, AttributeError):
                 pass
     except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)